fineract-provider/src/main/java/org/apache/fineract/scheduledjobs/service/ScheduledJobRunnerServiceImpl.java

```diff
@@ -443,43 +466,109 @@ public void updateTrialBalanceDetails() throws JobExecutionException {
                 BigDecimal closingBalance = null;
-                if(!CollectionUtils.isEmpty(closingBalanceData))
+                if (!CollectionUtils.isEmpty(closingBalanceData))
                     closingBalance = closingBalanceData.get(0);
-                if(CollectionUtils.isEmpty(closingBalanceData)) {
+                if (CollectionUtils.isEmpty(closingBalanceData)) {
                     closingBalance = BigDecimal.ZERO;
-                    for(TrialBalance row : tbRows) {
+                    for (TrialBalance row : tbRows) {
                         closingBalance = closingBalance.add(row.getAmount());
                         row.setClosingBalance(closingBalance);
                     }
                 } else {
-                    for(TrialBalance tbRow : tbRows) {
+                    for (TrialBalance tbRow : tbRows) {
                         closingBalance = closingBalance.add(tbRow.getAmount());
                         tbRow.setClosingBalance(closingBalance);
                     }
                 }
                 this.trialBalanceRepositoryWrapper.save(tbRows);
             }
         }
+    }
+
+    @Override
+    @CronTarget(jobName = JobName.UPDATE_LOAN_SUMMARY)
+    public void updateLoanSummaryDetails(@SuppressWarnings("unused") final Map<String, String> jobParameters) throws JobExecutionException {
+
+        final int threadPoolSize=Integer.parseInt(jobParameters.get("thread-pool-size"));
+        final String officeId = jobParameters.get("officeId");
+        final ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+        final Office office = this.officeRepository.findOne(Long.parseLong(officeId));
+
+        List<Callable<Object>> posters = new ArrayList<Callable<Object>>();
+
+        //Get the  total count of loans
+        Integer numberofLoans=loanReadPlatformService.retrieveNumberOfActiveLoans();
+
+        Integer batchSize = (int) Math.ceil(numberofLoans/ threadPoolSize);
+        Integer maxLoanId=0;
+        Integer offset=0;
+
+        do{
+            UpdateLoanSummaryPoster poster = (UpdateLoanSummaryPoster) this.applicationContext.getBean("updateLoanSummaryPoster");
+            poster.setLoanApplicationWriteService(loanApplicationWriteService);
+            poster.setTenant(ThreadLocalContextUtil.getTenant());
+            poster.setBatchSize(batchSize);
+            maxLoanId+=batchSize+1;
+            poster.setMaxLoanId(maxLoanId);
+            poster.setOffSet(offset);
+            offset+=batchSize;
+            poster.setOfficeHierachy(office.getHierarchy()+ "%");
+            posters.add(Executors.callable(poster));
+            numberofLoans-=batchSize;
+        }while(numberofLoans>=0);
+
+        try {
+            List<Future<Object>> responses = executorService.invokeAll(posters);
+            checkCompletion(responses);
+        } catch (InterruptedException e1) {
+            e1.printStackTrace();
```

Comment:
NOK, nope; never `catch` & `printStackTrace` ... either rethrow or at least `LOG.error("...oups...", e1)`. PS: https://issues.apache.org/jira/browse/FINERACT-696 - is anyone interested in helping with that?